src/popup/components/Modals/ClaimGiftCard.vue

```diff
@@ -0,0 +1,406 @@
+          return t('common.ok');
+        default:
+          return null;
+      }
+    });
+
+    function setMaxAmount() {
+      amount.value = max.value.isPositive() ? max.value.toString() : '0';
+    }
+
+    async function handleMainButtonClick() {
+      loading.value = true;
+      switch (step.value) {
+        case STEPS.initial:
+          setMaxAmount();
+          await store.dispatch('invites/claim', {
+            secretKey: props.secretKey,
+            recipientId: recipientId.value,
+            isMax: true,
+          });
+          step.value = STEPS.redeemFull;
+          break;
+        case STEPS.form:
+          await store.dispatch('invites/claim', {
+            secretKey: props.secretKey,
+            recipientId: recipientId.value,
+            amount: amount.value,
+            isMax: max.value.toString() === amount.value.toString(),
+          });
+          step.value = STEPS.redeemPartially;
+          break;
+        case STEPS.redeemPartially:
+        case STEPS.redeemFull:
+          props.resolve();
+          break;
+        default:
+          throw new Error(`Unknown step ${step.value}`);
+      }
+      loading.value = false;
+    }
+
+    const isDisabled = computed(() => {
+      switch (step.value) {
+        case STEPS.form:
+          return !amount.value || !!errors.value?.amount;
+        default:
+          return false;
+      }
+    });
+
+    const currencyFormatted = computed(() => getFormattedFiat(+amount.value));
+
+    onMounted(async () => {
+      await updateBalance();
+      if (balance.value.toNumber() === 0) {
+        step.value = STEPS.redeemFull;
+        isCardEmpty.value = true;
+      }
+      setInterval(updateBalance, 3000);
+    });
+
+    return {
+      AnimatedSpinner,
+      aeAccountsSelectOptions,
+      amount,
+      balance,
+      currencyFormatted,
+      errors,
+      fee,
+      handleMainButtonClick,
+      isCardEmpty,
+      isDisabled,
+      loading,
+      mainButtonText,
+      max,
+      PROTOCOL_AETERNITY,
+      recipientId,
+      setMaxAmount,
+      step,
+      STEPS,
+      updateBalance,
+    };
+  },
+});
+</script>
+
+<style lang="scss" scoped>
+@use '../../../styles/variables';
+@use '../../../styles/typography';
+
+.claim-gift-card {
+  :deep(.container) {
+    min-height: 480px;
+  }
+
+  .header {
+    text-align: center;
+    margin-bottom: 4px;
+
+    @extend %face-sans-19-medium;
+  }
+
+  .check-circle-icon {
+    width: 40px;
+    height: 40px;
+    color: #00d3a1;
+    display: flex;
+    align-self: center;
+    margin: 8px auto 4px auto;
+  }
+
+  .redeemed-amount {
+    display: flex;
+    justify-content: center;
+    color: variables.$color-white;
+
+    @extend %face-sans-16-medium;
+
+    line-height: 24px;
+
+    .gray {
+      opacity: 0.75;
+      margin-left: 4px;
+    }
+  }
+
+  .details-item {
+    display: flex;
+    justify-content: center;
+
+    .account-selector {
+      :deep(.truncate) {
```

Comment:
Parent components should not affect child component internals. So please avoid using `:deep` selector. If you want to make a child component to behave or look differently please use bool prop and update the child component itself.